### PR TITLE
Add MaxCount, MedianCount, MaxMinDelta, NUniqueDaysX, NMostCommonFrequency, IsFirstWeekOfMonth

### LINKED ISSUE #2534: Add MaxCount, MedianCount, MaxMinDelta, NUniqueDaysX, NMostCommonFrequency, IsFirstWeekOfMonth

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #2534

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py [hunk 1]
```diff
@@ -709,6 +718,391 @@ def test_ninf(self):
 
     def test_with_featuretools(self, es):
         transform, aggregation = find_applicable_primitives(self.primitive)
-        primitive_instantiate = self.primitive()
```
Comment (dvreed77) [inline featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py:712]: nice

File: featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py [hunk 2]
```diff
@@ -709,6 +718,391 @@ def test_ninf(self):
...
+        answer = primitive_func(data)
+        assert answer == 1
+
+        data = pd.Series([1, 1, 2, 3, 4, 4, 4])
+        primitive_func = self.primitive().get_function()
+        answer = primitive_func(data)
+        assert answer == 3
+
+    def test_skipna(self):
+        data = pd.Series([1, 1, 2, 3, 4, 4, np.nan, 5])
```
Comment (dvreed77) [inline featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py:753]: include True version here as well?

Comment (gsheni) [reply featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py:753]: True is the default, so it is tested in other tests

**Comment by rwedge** [comment]
Should mention pinning pandas in the release notes

File: featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py [hunk 3]
```diff
@@ -709,6 +718,391 @@ def test_ninf(self):
...
+        primitive_instance = self.primitive()
+        aggregation.append(primitive_instance)
+        valid_dfs(es, aggregation, transform, self.primitive.name.upper())
+
+
+class TestMaxMinDelta(PrimitiveTestBase):
+    primitive = MaxMinDelta
+    array = pd.Series([1, 1, 2, 2, 3, 4, 5, 6, 7, 8])
+
+    def test_max_min_delta(self):
```
Comment (dvreed77) [inline featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py:775]: Interested to see what happens with Inf values?

**Review by dvreed77 — APPROVED** [review]
Looks good besides checking infinite values in max-min primitive

**Pull request merged**